### PR TITLE
metric: add support for metric pagination by default

### LINKED ISSUE #4: Metric list is not listing all metrics
See https://bugs.launchpad.net/python-gnocchiclient/+bug/1607392

### PR DESCRIPTION
**Pull request by jd**
Fixes #4

### CODE REVIEW

**Comment by jd** [comment]
Pull-request updated, HEAD is now 6b875114440b6549e32aafd86012f3ec9a0b5f47

**Comment by jd** [comment]
Pull-request updated, HEAD is now 292e8724323f9e8d640fbe102a957a302b823a72

File: gnocchiclient/v1/metric.py [hunk 1]
```diff
@@ -38,11 +39,22 @@ def list(self, limit=None, marker=None, sorts=None):
         :param sorts: list of resource attributes to order by. (example
                       ["user_id:desc-nullslast", "project_id:asc"]
         :type sorts: list of str
-
+        :type follow_links_next: Whether to follow the next link (pagination)
+                                 even if limit is set.
```
Comment (chungg) [inline gnocchiclient/v1/metric.py]: can you provide a bit more details on what this option does? it seems strange and seems like the user has to know implementation details to understand what it does.

Comment (jd) [reply gnocchiclient/v1/metric.py]: thinking about it, I think it's useless. updated the patch.

**Comment by jd** [comment]
Pull-request updated, HEAD is now da647a9501e8d427c56905e5c979302af8fd632d

File: gnocchiclient/tests/functional/test_metric.py [hunk 2]
```diff
@@ -51,6 +51,26 @@ def test_delete_several_metrics(self):
...
+                     "-d granularity:1s,points:86400 %s" % apname)
+        # Create 10 metrics
+        for i in range(10):
+            self.gnocchi(
+                u'metric', params=u"create"
+                u" --archive-policy-name %s" % apname)
+
+        result = self.gnocchi(u'metric', params=u"list")
+        self.assertGreaterEqual(
+            len(self.parser.tables(result)[0]['values']), 10)
```
Comment (chungg) [inline gnocchiclient/tests/functional/test_metric.py]: wouldn't this always be >10? the limit is 1000 i think so your page link code is never really checked.

**Comment by jd** [comment]
Pull-request updated, HEAD is now 19a5442d18562a0668e4f23b4388a30593c9812d

**Comment by jd** [comment]
Pull-request updated, HEAD is now 98dc047ce3aba83d7074530b55835f435b568e0f

**Comment by jd** [comment]
what I can't create 1k metric, this ci sucks

**Comment by jd** [comment]
Pull-request updated, HEAD is now 1b32adf43e0776cbc1c839914d427ace9f3c4b24

**Comment by jd** [comment]
Pull-request updated, HEAD is now 0ab553ef451ee6ecf6b69cfc071bfbf14e1a0c31

**Comment by jd** [comment]
Pull-request updated, HEAD is now 7c3e50ee8bb02d9819d2ede8c75ae10b362547eb

**Comment by jd** [comment]
Pull-request updated, HEAD is now 148516238b6109fab392a84d7b01a8812a7a9b82

**Comment by jd** [comment]
Pull-request updated, HEAD is now 481fe473aab05e9683b84a4dc39430af1b849056

**Comment by jd** [comment]
Pull-request updated, HEAD is now 2df83b62e76c500b704f4359a1e2e02f88b3caba

**Pull request merged**